samples/net/socket_echo/src/socket_echo.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+
+#ifndef __ZEPHYR__
+
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+
+#define init_net()
+
+#else
+
+#include <net/socket.h>
+#include <kernel.h>
+#include <net_sample_app.h>
+
+void init_net(void)
+{
+	int ret = net_sample_app_init("socket_echo", NET_SAMPLE_NEED_IPV4,
+				      K_SECONDS(3));
+
+	if (ret < 0) {
+		printf("Application init failed\n");
+		k_panic();
+	}
+}
+
+#endif
+
+int main(void)
+{
+	int serv;
+	struct sockaddr_in bind_addr;
+
+	init_net();
+
+	serv = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+
+	bind_addr.sin_family = AF_INET;
```

Comment:
So (also partially answering @lpereira's comment below), the primary purpose of this sample was to show that the same code (modulo includes of course) can be used both on POSIX and Zephyr. Turns out, I missed to include "Makefile.posix" to make that obvious, I'll add it. Then, Linux for example supports "dual stack" sockets which can be accessed via IPv4 or IPv6, but I don't think we have that feature in Zephyr. Otherwise, to support IPv4 and IPv6 *at the same time*, there would be needed 2 sockets, and poll()/select(), which are scheduled so "stage 3" of Sockets implementation (this patch being stage 1).

Perhaps you mean *either IPv4 or IPv6*? That can be done, though again, the idea was to show how sockets can be portable between POSIX and Zephyr, without extra #ifdef noise.